projects/experimental/directives/button-close/button-close.directive.ts

```diff
@@ -0,0 +1,23 @@
+import {Directive, Inject} from '@angular/core';
+import {TuiStringHandler} from '@taiga-ui/cdk';
+import {tuiButtonOptionsProvider} from '@taiga-ui/experimental/components';
+import {TUI_ICON_RESOLVER} from '@taiga-ui/experimental/tokens';
+
+import {TUI_BUTTON_CLOSE_ICON} from './button-close.options';
+
+@Directive({
+    selector: '[tuiIconButton][tuiButtonClose]',
+    providers: [tuiButtonOptionsProvider({appearance: 'neutral', size: 's'})],
+    host: {
+        '[style.--t-mask-left]': '"url(" + resolver(icon) + ")"',
```

Comment:
We need to keep an eye on this when we move icons directive to hostDirectives so that there's no conflict. Maybe add a TODO?